server/utils/file/csvReader.js

```diff
@@ -7,88 +7,70 @@ import * as StringUtils from '@core/stringUtils'
 export const createReaderFromStream = (stream, onHeaders = null, onRow = null, onTotalChange = null) => {
 
   let canceled = false
-  const queue = new Queue()
 
   const start = () => new Promise((resolve, reject) => {
-    let ended = false
+    const queue = new Queue()
     let headers = null
     let total = 0
-    let processingRow = false //prevents the call to processNext when a row is already being processed
 
     /**
      * Executes the specified function fn in a try catch.
      * Calls "reject" if the execution throws an error.
      */
-    const _tryOrCancel = async fn => {
+    const _tryOrCancel = async fnPromise => {
       try {
-        await fn()
+        await fnPromise
       } catch (e) {
         cancel()
         reject(e)
       }
     }
 
-    const processNext = () => {
+    const _indexRowByHeaders = row => headers
+      ? headers.reduce(
+        (accRow, header, index) => Object.assign(accRow, { [header]: StringUtils.trim(row[index]) }),
+        {}
+      )
+      : row
+
```

Comment:
Would R.indexBy be more readable? trim would have to be done separately though

Reply:
R.indexBy is not suitable in this context